types/expo/index.d.ts

```diff
@@ -1010,150 +1000,479 @@ export namespace Constants {
+     */
+    interface Image {
+        uri: string;
+        /**
+         * iOS only
+         * In Android you can get dimensions using `ReactNative.Image.getSize`
+         */
+        width?: number;
+        /**
+         * iOS only
+         * In Android you can get dimensions using `ReactNative.Image.getSize`
+         */
+        height?: number;
+        /**
+         * Avoid using Base 64 in React Native
+         * iOS only
+         */
+        base64?: string;
+    }
+
+    /**
+     * A parent to contacts. A contact can belong to multiple groups.
+     * iOS Only
+     */
+    interface Group {
+        id: string;
+        name: string;
+    }
+
+    /**
+     * A parent to contacts and groups.
+     * iOS Only
+     */
+    interface Container {
+        id: string;
+        name: string;
+    }
+
+    /**
+     * Denotes the functionality of a native contact form
+     */
+    interface FormOptions {
+        displayedPropertyKeys?: Field[];
+        message?: string;
+        alternateName?: string;
+        cancelButtonTitle?: string;
+        groupId?: string;
+        allowsEditing?: boolean;
+        allowsActions?: boolean;
+        shouldShowLinkedContacts?: boolean;
+        isNew?: boolean;
+        preventAnimation?: boolean;
+    }
+
+    /**
+     * Used to query contacts from the user's device
+     */
+    interface ContactQuery {
+        fields?: Field[];
+        pageSize?: number;
+        pageOffset?: number;
+        id?: string;
+        /** Android Only */
+        sort?: SortType;
+        /** iOS Only */
+        name?: string;
+        /** iOS Only */
+        groupId?: string;
+        /** iOS Only */
+        containerID?: string;
+        /** iOS Only */
+        rawContacts?: boolean;
+    }
+
+    interface ContactResponse {
         data: Contact[];
-        total: number;
+        /**
+         * This will be true if there are more contacts to retrieve beyond what is returned
+         */
         hasNextPage: boolean;
+        /**
+         * `true` if there are previous contacts that weren't retrieved due to `pageOffset`
+         */
         hasPreviousPage: boolean;
     }
 
-    function getContactsAsync(options: Options): Promise<Response>;
-    function getContactByIdAsync(options: { id?: string; fields?: FieldType[] }): Promise<Contact>;
+    /**
+     * Used to query native contact groups.
+     * iOS Only
+     */
+    interface GroupQuery {
+        groupName?: string;
+        groupId?: string;
+        containerId?: string;
+    }
+
+    /**
+     * Used to query native contact containers.
+     * iOS Only
+     */
+    interface ContainerQuery {
+        contactId?: string;
+        groupId?: string;
+        containerId?: string;
+    }
+
+    // Methods
+
+    /**
+     * Return a list of contacts that fit a given criteria. You can get all of the contacts by passing no criteria.
+     */
+    function getContactsAsync(contactQuery?: ContactQuery): Promise<ContactResponse>;
```

Comment:
As much as [I can see](https://docs.expo.io/versions/latest/sdk/contacts#example), the method signature is like the one you deleted, isn't?

Reply:
I was going through the [source code](https://github.com/expo/expo/blob/sdk-31/packages/expo-contacts/src/Contacts.js) before; this parameter is optional. But sure enough, typings should match the docs, I just missed the fact that docs don't say it's optional. I'll fix that